src/friendly-iframe-embed.js

```diff
@@ -738,19 +782,17 @@ export class FriendlyIframeEmbed {
       })
     );
   }
-
   /**
-   * Install extensions in the child window (friendly iframe). The pre-install
-   * callback, if specified, is executed after polyfills have been configured
-   * but before the first extension is installed.
+   * Prepare for installing extensions in the child window (friendly iframe).
+   * The pre-install callback, if specified, is executed after polyfills have been configured.
    * @param {!./service/extensions-impl.Extensions} extensions
    * @param {!Window} childWin
    * @param {!Array<string>} extensionIds
    * @param {function(!Window, ?./service/ampdoc-impl.AmpDoc=)=} opt_preinstallCallback
    * @return {!Promise}
    * @visibleForTesting
    */
-  installExtensionsInChildWindow(
+  preInstallExtensionsInChildWindow(
```

Comment:
nit: stubExtensions...?

Reply:
It also does polyfilling and service registration. Maybe we should leave this name?

Reply:
I just found it a bit confusing that preload actually happens in `doInstall...` maybe just add to jsdoc about this installs polyfills, global styles, services ...?

Reply:
`preInstall` is chained with `doInstall` by promise, so they are two separate methods.

https://github.com/ampproject/amphtml/pull/27585/files#diff-3322ec57b1a7fa01a69aa4748f98e619R278

Reply:
Right, can you just add some more detail to the jsdoc for both so its clear preinstall does X,Y,Z and doInstall does A,B,C without reading all the logic?